packages/nimble/inst/tests/test_utils.R

```diff
@@ -26,6 +26,49 @@ temporarilyAssignInGlobalEnv <- function(value) {
     do.call('on.exit', list(rmCommand), envir = parent.frame())
 }
 
+withTempProject <- function(code) {
+    code <- substitute(code)
+    project <- nimble:::nimbleProjectClass()
+    nimbleOptions(nimbleProject = project)
+    on.exit({
+        ## messages are suppressed by test_that, so "assign('.check', 1, globalenv())" can be used as a way to verify this code was called 
+        .project <- nimbleOptions()$nimbleProject
+        nimbleOptions(nimbleProject = NULL) ## clear this before clearCompiled step, in case clearCompiled() itself fails
+        .project$clearCompiled()
+    })
+    try(eval(code, envir = parent.frame()))
+}
+
+expect_compiles <- function(...) {
+    nimbleOptions(stopCompilationBeforeLinking = TRUE)
+    nimbleOptions(forceO1 = TRUE)
+    on.exit({
+        assign('.check', 1, globalenv())
+        nimbleOptions(stopCompilationBeforeLinking = FALSE)
+        nimbleOptions(forceO1 = FALSE)
+    })
+    ans <- try(compileNimble(...)) ## expecting a thrown error
```

Comment:
I guess I don't understand the R exception mechanism, but couldn't you simply
```
    })
    compileNimble(...)
}
```
so that if `compileNimble` fails (via stop()), that will propagate up?